frontend/src/components/HomeHeader/index.tsx

```diff
@@ -0,0 +1,25 @@
+import { dehydrate, QueryClient, useQuery } from '@tanstack/react-query';
+import { getTotalArticleCount } from '@/api/article';
+
+import * as S from './index.styles';
+
+const HomeHeader = () => {
+  const { data, isSuccess } = useQuery<{ count: number }>(['article-count'], getTotalArticleCount);
+  if (isSuccess) {
+    return <S.Title>{data.count}개의 계획이 모여있어요</S.Title>;
+  }
+  return <S.Title>0개의 계획이 모여있어요</S.Title>;
+};
+
+export async function getStaticProps() {
+  const queryClient = new QueryClient();
+  await queryClient.prefetchQuery(['article-count'], getTotalArticleCount);
+
+  return {
+    props: {
+      dehydratedState: dehydrate(queryClient),
+    },
+  };
+}
```

Comment:
이건 여기 컴포넌트 파일에서보다는 별도의 파일로 빼는 편이 좋지 않을까요?.?

Reply:
이게 Next를 제대로 이해한건지는 모르겠는데, 일단은 설명을 읽어보았을 때에는 적용되는 컴포넌트에 두라는 것 같긴 합니다 하하 
https://nextjs.org/docs/basic-features/data-fetching/get-static-props

근데 뭔지 모르겠어요

Reply:
오호 그렇네요
그럼 getStaticProps이랑 getServerSideProps 중에서 getStaticProps를 사용하신 이유가 있나요? getStaticProps를 사용하면 빌드 시에만 데이터를 가져오고 빌드가 되면 변하지 않는 정적 페이지가 된다고 하고, getServerSideProps는 페이지가 요청 시마다 재요청이 된다는데 오늘 얘기했던 최신 순으로 불러온다고 치면 getServerSideProps가 좀 더 적절하지 않을까 하는 생각이 살짝 들어서요! (제가 제대로 이해한 건지 모르겠네요 머쓱)

Reply:
하리 의견에 동의합니다! 여기서는 SSG가 아닌 SSR을 사용해야 할 것 같네유

Reply:
저도 그렇게 이해했는데, React query 공식문서에세는 예시를 StaticProps을 사용하더라고요 
PR 링크에 첨부해둔 React-Query의 예시 코드에 적혀있습니다! 
getStaticProps: It will cause a second hydration by React Query
hydrate 현상을 활용하려면 해당 함수를 사용하라고 하더라고요

Reply:
아니면 하리가 고쳐준 msw의 require문제일지도...?